react/features/calendar-sync/web/microsoftCalendar.js

```diff
@@ -0,0 +1,564 @@
+
+    // Get the number of seconds the token is valid for, subtract 5 minutes
+    // to account for differences in clock settings and convert to ms.
+    const expiresIn = (parseInt(params.get('expires_in'), 10) - 300) * 1000;
+    const now = new Date();
+    const expireDate = new Date(now.getTime() + expiresIn);
+
+    params.set('tokenExpires', expireDate.getTime().toString());
+
+    // FIXME: what does this do?
+    return Array.from(params.entries())
+        .reduce((main, [ key, value ]) => {
+            return {
+                ...main,
+                [key]: value
+            };
+        }, {});
+}
+
+/**
+ * Converts the parameters from a Microsoft auth redirect into an object of
+ * token parts. The value "null" will be returned if the params do not produce
+ * a valid token.
+ *
+ * @param {Object} tokenInfo - The token object.
+ * @param {Object} guids - The guids for authState and authNonce that should
+ * match in the token.
+ * @param {Object} appId - The Microsoft application this token is for.
+ * @private
+ * @returns {Object|null}
+ */
+function getValidatedTokenParts(tokenInfo, guids, appId) {
+    // Make sure the token matches the request source by matching the GUID.
+    if (tokenInfo.state !== guids.authState) {
+        return null;
+    }
+
+    const idToken = tokenInfo.id_token;
+
+    // A token must exist to be valid.
+    if (!idToken) {
+        return null;
+    }
+
+    const tokenParts = idToken.split('.');
+
+    if (tokenParts.length !== 3) {
+        return null;
+    }
+
+    const payload
+         = rs.KJUR.jws.JWS.readSafeJSONString(rs.b64utoutf8(tokenParts[1]));
+
+    if (payload.nonce !== guids.authNonce
+        || payload.aud !== appId
+        || payload.iss
+            !== `https://login.microsoftonline.com/${payload.tid}/v2.0`) {
+        return null;
+    }
+
+    const now = new Date();
+
+    // Adjust by 5 minutes to allow for inconsistencies in system clocks.
+    const notBefore = new Date((payload.nbf - 300) * 1000);
+    const expires = new Date((payload.exp + 300) * 1000);
+
+    if (now < notBefore || now > expires) {
+        return null;
+    }
+
+    return {
+        accessToken: tokenInfo.access_token,
+        idToken,
+        tokenExpires: expires.getTime(),
+        userDisplayName: payload.name,
+        userSigninName: payload.preferred_username,
+        userDomainType:
+            payload.tid === MS_API_CONFIGURATION.MS_CONSUMER_TENANT
+                ? 'consumers' : 'organizations'
+    };
+}
+
+/* eslint-disable max-params */
+/**
+ * Retrieves calendar entries from a specific calendar.
+ *
+ * @param {Object} client - The Microsoft-graph-client initialized.
+ * @param {string} calendarId - The calendar ID to use.
+ * @param {number} fetchStartDays - The number of days to go back
+ * when fetching.
+ * @param {number} fetchEndDays - The number of days to fetch.
+ * @returns {Promise<any> | Promise}
+ * @private
+ */
+function requestCalendarEvents(
```

Comment:
the eslint disabling can be done on this line, I think, and using "eslint-disable-line" instead of "eslint-disable" plus "eslint-enable".